File: scripts/splice.js
// splice.js <replacement-file> <start-line> <end-line>
// Replaces lines start..end in src/App.jsx with content from replacement-file
var fs = require('fs');
var path = require('path');
var repFile = process.argv[2];
var start = parseInt(process.argv[3], 10);
var end = parseInt(process.argv[4], 10);

var appPath = path.join(__dirname, '..', 'src', 'App.jsx');
var lines = fs.readFileSync(appPath, 'utf-8').split('\n');
var replacement = fs.readFileSync(repFile, 'utf-8').split('\n');

// Remove trailing empty line if present
if (replacement[replacement.length - 1] === '') {
  replacement.pop();
}

console.log('Replacing lines ' + start + '-' + end + ' (' + (end - start + 1) + ' lines) with ' + replacement.length + ' new lines');
console.log('Old line ' + start + ': ' + lines[start - 1].substring(0, 60));
console.log('New line 1: ' + replacement[0].substring(0, 60));

var before = lines.slice(0, start - 1);
var after = lines.slice(end);
var newContent = before.concat(replacement, after).join('\n');
fs.writeFileSync(appPath, newContent, 'utf-8');
console.log('Done! File updated.');

File: scripts/splice.py
"""splice.py <replacement-file> <start-pattern> <end-pattern>
Reads App.jsx, finds start-pattern and end-pattern lines, replaces everything between them (exclusive) with the content of replacement-file.
"""
import sys, re

replacement_file = sys.argv[1]
start_pattern = sys.argv[2]
end_pattern = sys.argv[3]

with open(replacement_file, 'r', encoding='utf-8') as f:
    replacement = f.read()

with open('src/App.jsx', 'r', encoding='utf-8') as f:
    lines = f.readlines()

start_idx = None
end_idx = None
for i, line in enumerate(lines):
    if start_pattern in line and start_idx is None:
        start_idx = i
    if start_pattern in line and start_idx is not None:
        # continue searching for end after start
        pass
    if end_pattern in line and start_idx is not None and i > start_idx:
        end_idx = i
        break

if start_idx is None or end_idx is None:
    print(f"ERROR: start={start_idx} end={end_idx}")
    sys.exit(1)

print(f"Replacing lines {start_idx+1}-{end_idx+1} ({end_idx - start_idx + 1} lines)")

new_lines = lines[:start_idx] + [replacement + '\n'] + lines[end_idx+1:]

with open('src/App.jsx', 'w', encoding='utf-8') as f:
    f.writelines(new_lines)

print("Done!")
